### PR TITLE
Core: Deprecate `newOutputFile` from `{ClusteredWriter,FanoutWriter}`

### PR DESCRIPTION
**Pull request by Fokko**
They are not used anymore

### CODE REVIEW

**Comment by Fokko** [comment]
https://github.com/apache/iceberg/pull/7685#issuecomment-1579536134

File: core/src/main/java/org/apache/iceberg/io/ClusteredWriter.java [hunk 1]
```diff
@@ -132,6 +132,8 @@ public final R result() {
     return aggregatedResult();
   }
 
+  /** @deprecated will be removed in 1.4.0 */
```
Comment (aokolnychyi) [inline core/src/main/java/org/apache/iceberg/io/ClusteredWriter.java:135]: Given that 1.3.0 is out, should it be deprecated since 1.4.0, removed in 1.5.0?

**Review by aokolnychyi — APPROVED** [review]
LGTM. One question about the deprecation notice. Merge whenever you are ready.

File: core/src/main/java/org/apache/iceberg/io/ClusteredWriter.java (hunk 1 above)
Comment (Fokko) [reply core/src/main/java/org/apache/iceberg/io/ClusteredWriter.java:135]:
```suggestion
  /** @deprecated will be removed in 1.5.0 */
```

File: core/src/main/java/org/apache/iceberg/io/FanoutWriter.java [hunk 2]
```diff
@@ -95,6 +95,8 @@ public final R result() {
     return aggregatedResult();
   }
 
+  /** @deprecated will be removed in 1.4.0 */
```
Comment (Fokko) [inline core/src/main/java/org/apache/iceberg/io/FanoutWriter.java:98]:
```suggestion
  /** @deprecated will be removed in 1.5.0 */
```

File: core/src/main/java/org/apache/iceberg/io/ClusteredWriter.java (hunk 1 above)
Comment (Fokko) [reply core/src/main/java/org/apache/iceberg/io/ClusteredWriter.java]: Indeed, this was copy pasted 🤦🏻

**Pull request merged**